### PR TITLE
Upgrade to Django 1.9 (update PR #2435)

### PR DESCRIPTION
**Pull request by ericholscher**
This updates #2435, which seems to have disappeared.

Things still needed to do:

- [x] Run local tests to make sure things work in real builds

### CODE REVIEW

File: readthedocs/rtd_tests/tests/test_privacy_urls.py [hunk 1]
```diff
@@ -59,6 +59,9 @@ def assertResponse(self, path, name=None, method=None, data=None, **kwargs):
         if self.context_data and getattr(response, 'context'):
             self._test_context(response)
         for (key, val) in response_attrs.items():
+            if getattr(response, key) != val:
+                from pdb import set_trace
```
Comment (travcunn) [inline readthedocs/rtd_tests/tests/test_privacy_urls.py]: Debugging code left in

File: readthedocs/core/utils/extend.py [hunk 2]
```diff
@@ -3,7 +3,8 @@
 import inspect
 
 from django.conf import settings
-from django.utils.module_loading import import_by_path
+from django.utils.module_loading import import_string
+from django.utils.functional import LazyObject
```
Comment (berkerpeksag) [inline readthedocs/core/utils/extend.py]: Unused import.

File: requirements/pip.txt [hunk 3]
```diff
@@ -8,14 +8,14 @@ sphinx_rtd_theme==0.2.0
 Pygments==2.0.2
 mkdocs==0.14.0
 readthedocs-build==2.0.6
-django==1.8.16
+django==1.9.0
```
Comment (berkerpeksag) [inline requirements/pip.txt]: 1.9.12

**Comment by ericholscher** [comment]
Ran a few local builds, and things look good

**Comment by ericholscher** [comment]
Running into an issue with the build state not updating correctly. Looking into it.

**Comment by ericholscher** [comment]
Got it working. `total_seconds()` was returning a non-int, which broke the API :)

**Comment by ericholscher** [comment]
Calling this Ready for Review -- it is passing on local builds, and looks good on my side.

File: readthedocs/projects/admin.py [hunk 4]
```diff
@@ -48,18 +47,21 @@ class DomainInline(admin.TabularInline):
...
 
-    def view_ratio(self, instance):
-        return instance.view_ratio * 100
+#     def view_ratio(self, instance):
+#         return instance.view_ratio * 100
 
-    def click_ratio(self, instance):
-        return instance.click_ratio * 100
+#     def click_ratio(self, instance):
+#         return instance.click_ratio * 100
```
Comment (agjohnson) [inline readthedocs/projects/admin.py]: Can we patch this admin view from the donate app?

File: readthedocs/comments/models.py [hunk 5]
```diff
@@ -111,6 +111,7 @@ class DocumentNodeSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = DocumentNode
+        exclude = ('')
```
Comment (berkerpeksag) [inline readthedocs/comments/models.py]: Looking at http://www.django-rest-framework.org/api-guide/serializers/#specifying-which-fields-to-include isn't this actually an equivalent of `fields = '__all__'`?

File: readthedocs/doc_builder/environments.py [hunk 6]
```diff
@@ -401,7 +402,7 @@ def update_build(self, state=None):
 
         if self.start_time:
             build_length = (datetime.utcnow() - self.start_time)
-            self.build['length'] = build_length.total_seconds()
+            self.build['length'] = int(build_length.total_seconds())
```
Comment (berkerpeksag) [inline readthedocs/doc_builder/environments.py]: Out of curiosity, what was the return type of `build_length.total_seconds()`?

File: readthedocs/comments/models.py (hunk 5 above)
Comment (ericholscher) [inline readthedocs/comments/models.py]: Yea, but that was giving me an error when I was extending the fields list with custom fields on the Serializer.

File: readthedocs/doc_builder/environments.py (hunk 6 above)
Comment (ericholscher) [inline readthedocs/doc_builder/environments.py]: It was returning subsecond precision :)

File: readthedocs/projects/admin.py (hunk 4 above)
Comment (ericholscher) [inline readthedocs/projects/admin.py]: Good call, will give it a try.

**Comment by ericholscher** [comment]
@agjohnson I was doing all my conda debugging and other local dev on this branch yesterday, and it seemed to be working fine. I'm not sure if it will get any real testing until we push it to production, so I'm not sure what we really gain from not merging it. Are you thinking about hotfixing it into prod and keeping it off master?

**Comment by ericholscher** [comment]
Ah, you're suggesting merging it into a `dev` branch for a couple weeks, and doing mainline dev in there, to work out the kinks?

**Comment by agjohnson** [comment]
Okay, yeah that's fine. I don't have a strong opinion here either. We don't have a good pattern for this, and yeah, the most testing will come with prod deploy.

File: readthedocs/urls.py [hunk 7]
```diff
@@ -80,8 +75,14 @@
 
 # Export URLs
 groups = [basic_urls, rtd_urls, project_urls, api_urls, core_urls, i18n_urls,
-          money_urls, deprecated_urls]
-
+          deprecated_urls]
+
+if getattr(settings, 'INCLUDE_DONATIONS', True):
```
Comment (agjohnson) [inline readthedocs/urls.py]: I feel like this is an unneeded pattern that is replicating part of what ``INSTALLED_APPS`` should provide

Comment (ericholscher) [inline readthedocs/urls.py]: So we can do `if 'donate' in INSTALLED_APPS` instead?

**Comment by ericholscher** [comment]
The test fail is stupid appdirs stuff. Going to merge this and get it deployed today. 🎆 

Thanks to @travcunn for the original patch!

**Pull request merged**